src/matlab2tikz.m

```diff
@@ -3573,10 +3586,19 @@ case size(cData,2) % specific y-data is given
     % plot the thing
     xData = get(h, 'XData');
     yData = get(h, 'YData');
-    [m2t, table, tabOpts] = makeTable(m2t, '', xData, '', yData);
+    if m2t.axesContainers{end}.is3D
```

Comment:
Please also either take `hgtransform` into account here or add a comment `%TODO: account for hgtransform`